staging/src/k8s.io/client-go/applyconfigurations/meta/v1/objectmeta.go

```diff
@@ -0,0 +1,147 @@
+	ManagedFields              *[]ManagedFieldsEntryApplyConfiguration `json:"managedFields,omitempty"`
+}
+
+// ObjectMetaApplyConfiguration constructs an declarative configuration of the ObjectMeta type for use with
+// apply.
+func ObjectMeta() *ObjectMetaApplyConfiguration {
+	return &ObjectMetaApplyConfiguration{}
+}
+
+// SetName sets the Name field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetName(value string) *ObjectMetaApplyConfiguration {
+	b.Name = &value
+	return b
+}
+
+// SetGenerateName sets the GenerateName field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetGenerateName(value string) *ObjectMetaApplyConfiguration {
+	b.GenerateName = &value
+	return b
+}
+
+// SetNamespace sets the Namespace field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetNamespace(value string) *ObjectMetaApplyConfiguration {
+	b.Namespace = &value
+	return b
+}
+
+// SetSelfLink sets the SelfLink field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetSelfLink(value string) *ObjectMetaApplyConfiguration {
+	b.SelfLink = &value
+	return b
+}
+
+// SetUID sets the UID field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetUID(value types.UID) *ObjectMetaApplyConfiguration {
+	b.UID = &value
+	return b
+}
+
+// SetResourceVersion sets the ResourceVersion field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetResourceVersion(value string) *ObjectMetaApplyConfiguration {
+	b.ResourceVersion = &value
+	return b
+}
+
+// SetGeneration sets the Generation field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetGeneration(value int64) *ObjectMetaApplyConfiguration {
+	b.Generation = &value
+	return b
+}
+
+// SetCreationTimestamp sets the CreationTimestamp field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetCreationTimestamp(value v1.Time) *ObjectMetaApplyConfiguration {
+	b.CreationTimestamp = &value
+	return b
+}
+
+// SetDeletionTimestamp sets the DeletionTimestamp field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetDeletionTimestamp(value v1.Time) *ObjectMetaApplyConfiguration {
+	b.DeletionTimestamp = &value
+	return b
+}
+
+// SetDeletionGracePeriodSeconds sets the DeletionGracePeriodSeconds field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetDeletionGracePeriodSeconds(value int64) *ObjectMetaApplyConfiguration {
+	b.DeletionGracePeriodSeconds = &value
+	return b
+}
+
+// SetLabels sets the Labels field in the declarative configuration to the given value.
+func (b *ObjectMetaApplyConfiguration) SetLabels(value map[string]string) *ObjectMetaApplyConfiguration {
```

Comment:
do we really want them to pass in that map? or do we want to help them build it?

Reply:
I just added some comments to https://github.com/kubernetes/kubernetes/pull/99012#pullrequestreview-593764126, (particularly https://github.com/kubernetes/kubernetes/pull/99012#discussion_r579340949) with some more details about if/how to build maps. Curious what you think of using varargs vs. introducing map/list typed and adding utility functions to try to make the build code more convenient.